docs/root/faq/websocket.rst

```diff
@@ -0,0 +1,19 @@
+How does Envoy handle Upgrades (especially WebSocket)?
+======================================================
+
+
+Envoy currently supports two modes of Upgrade behavior.
+
+The first is the deprecated WebSocket-only upgrade path. This code path works for HTTP/1 downstream
+and upstreams. It detects requests with "Upgrade: websocket" headers and upgrades from HTTP processing
+mode to TCP proxy mode, establishing a raw TCP connection upstream, forwarding the raw headers, any
+HTTP body, and WebSocket payload unmodified by the HTTP filter chain.
+
```

Comment:
just a note: if nobody other than Istio and cloud foundry are using the websocket upgrade stuff, I would be happy to drop this explicit "allow_websocket_upgrade: true" immediately. Its easy for us to configure the filter chains for websockets (and CF folks would do the same as they use Pilot code)

Reply:
As tempted as I am to clean up code early, I don't think there's a way to poll all websocket users, so I plan on the normal deprecation timeline :-)

Reply:
Oh and worth noting, you may need timeouts to work properly before switching over.    As part of making the tests old-and-new-style compatible I discovered Envoy doesn't have proper request timeouts and due to the bidi nature of websockets and the way current Envoy timers work, they'll linger forever :-/

I believe @htuch was planning on picking up https://github.com/envoyproxy/envoy/issues/3654 

If that blocks you from moving over I'll make sure he or I pick it up in the next 1-2 weeks since I'd really like the new code field tested in prod before I deprecate the old code.

Reply:
@alyssawilk I'm working on #1778 rather than #3654.

Reply:
Ah noted.   Thankfully that'll work just as well :-)